.circleci/config.yml

```diff
@@ -0,0 +1,183 @@
+      - checkout
+      - run: pip install -e ./
+      - run: mbed --version
+      - run: py.test test
+      - run: mbed toolchain -G GCC_ARM
+      - run: mbed target -G K64F
+      - run: mbed config -G protocol ssh
+      - run: mkdir .tests
+      - run: cd .tests && mbed new new-test
+      - run: cd .tests/new-test && mbed ls
+      - run: cd .tests/new-test && mbed releases -r
+      - run: cd .tests/new-test && mbed compile --source=. --source=mbed-os/TESTS/integration/basic -j 0
+      - run: cd .tests/new-test && mbed test --compile -n mbed-os-tests-integration-basic -j 0
+      - run: cd .tests && mbed import https://developer.mbed.org/teams/Morpheus/code/mbed-Client-Morpheus-hg hg-test
+      - run: cd .tests/hg-test && mbed ls
+      - run: cd .tests/hg-test && mbed releases -r
+      - run: cd .tests/hg-test && mbed update b02527cafcde8612ff051fea57e9975aca598807 --clean
+      - run: cd .tests/hg-test && mbed update --clean
+      - run: cd .tests/hg-test && mbed compile -j 0
+      - run: cd .tests && mbed import https://developer.mbed.org/users/samux/code/USBSerial_HelloWorld bld-test
+      - run: cd .tests/bld-test && mbed ls
+      - run: cd .tests/bld-test && mbed releases -r
+      - run: cd .tests/bld-test/mbed && mbed update 85 --clean
+      - run: cd .tests/bld-test && mbed update --clean
+      - run: cd .tests/bld-test && mbed compile -m LPC1768 -j 0
+      - run: cd .tests && mbed new supported-tests
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.6.0"
+          mbed update --clean "mbed-os-5.6.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.5.0"
+          mbed update --clean "mbed-os-5.5.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.4.0"
+          mbed update --clean "mbed-os-5.4.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S 
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.3.0"
+          mbed update --clean "mbed-os-5.3.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.2.0"
+          mbed update --clean "mbed-os-5.2.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S
+      - run: |-
+          cd .tests/supported-tests
+          echo "Testing mbed-os-5.1.0"
+          mbed update --clean "mbed-os-5.1.0"
+          mbed compile -S
+          mbed toolchain -S
+          mbed target -S
+          mbed export -S
```

Comment:
Instead of copy and pasting these commands, can the common ones be extracted to either a script or circleci job?

Reply:
Indeed. I also think there should be one test per feature release of mbed-os, but that was out of scope for this PR.